### PR TITLE
VZ-2129 & VZ-2198: Update examples to remove hosts from app config

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

This PR covers VZ-2129 and VZ-2198, updating the example apps. Highlights include:
- Remove hosts from app configs
- Update the e2e example tests to handle the dynamic host names
- Update READMEs to reflect above, also use TLS in the examples

I have yet to confirm the TLS usage in the READMEs, but will do that next.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by markxnelson — DISMISSED** [review]

**Review by rosemarymarano — COMMENTED** [review]
A few minor edits. One thing I noticed is that all the examples, except Bob's Books, have the copyright info at the bottom of the page. Perhaps, we should add it to Bob's Books to be consistent. Thank you.

File: examples/bobs-books/README.md [hunk 1]
```diff
@@ -20,13 +20,13 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
 1. Create a namespace for the example and add a label identifying the namespace as managed by Verrazzano.
 
     ```
-    kubectl create namespace bobs-books
-    kubectl label namespaces bobs-books verrazzano-managed=true
+    $ kubectl create namespace bobs-books
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md:23]: Fixed.

File: examples/bobs-books/README.md [hunk 2]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
 
 1. Apply the example resources to deploy the application.
    ```
-   kubectl apply -f .
+   $ kubectl apply -f .
    ```
 
 1. Wait for all of the pods in the Bob's Books example application to be ready.
    You may need to repeat this command several times before it is successful.
    The WebLogic server and Coherence pods can take some time to be created and `Ready`.
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md]: Fixed.

File: examples/bobs-books/README.md [hunk 3]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
    ```
-   kubectl wait --for=condition=Ready pods --all -n bobs-books --timeout=600s
+   $ kubectl wait --for=condition=Ready pods --all -n bobs-books --timeout=600s
    ```
+
 1. Get the `EXTERNAL_IP` address of the istio-ingressgateway service.
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md]: Fixed.

File: examples/bobs-books/README.md [hunk 4]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
+   $ kubectl get gateway robert-ingress-rule-0-gw -n bobs-books -o jsonpath={.items[0].spec.servers[0].hosts[0]}
+   robert.bobs-books.11.22.33.44.xip.io
+
+   $ kubectl get gateway bobby-ingress-rule-0-gw -n bobs-books -o jsonpath={.items[0].spec.servers[0].hosts[0]}
+   bobby-front-end.bobs-books.11.22.33.44.xip.io
+   ```
+
 1. Access the applications. To access the applications in a browser, you will need to do one of the following:
-    * **Option 1:** Temporarily modify the `/etc/hosts` file (on Mac or Linux) or `c:\Windows\System32\Drivers\etc\hosts` file (on Windows 10), to add entries mapping the hosts used by the applications to the external IP address assigned to your gateway. For example:
+    * **Option 1:** If you are using `xip.io` you can just access the applications using the generated host names, for example:
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md]: FIxed.

File: examples/bobs-books/README.md [hunk 5]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
 
-      b. Bobby's Books UI at `http://bobbys-books.example.com/bobbys-front-end`.
+      b. Bobby's Books UI at `https://bobbys-books.example.com/bobbys-front-end`.
 
-      c. Bob's order manager  UI at `http://bobs-orders.example.com/bobs-bookstore-order-manager/orders`.
+      c. Bob's order manager  UI at `https://bobs-orders.example.com/bobs-bookstore-order-manager/orders`.
 
-    * **Option 2:** Alternatively, point your own DNS name to the load balancer's external IP address. In this case, you would need to have edited the `bobs-books-app.yaml` file to use the appropriate values under the `hosts` section for each application (such as `your-roberts-books-host.your.domain`), before deploying the applications.
+    * **Option 3:** Alternatively, point your own DNS name to the load balancer's external IP address. In this case, you would need to have edited the `bobs-books-app.yaml` file to use the appropriate values under the `hosts` section for each application (such as `your-roberts-books-host.your.domain`), before deploying the applications.
       Then, you can use a browser to access each of the applications as shown below:
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md:118]: Fixed.

File: examples/bobs-books/README.md [hunk 6]
```diff
@@ -20,13 +20,13 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
-    kubectl create namespace bobs-books
-    kubectl label namespaces bobs-books verrazzano-managed=true
+    $ kubectl create namespace bobs-books
+    $ kubectl label namespaces bobs-books verrazzano-managed=true
     ```
 
 1. Create a `docker-registry` secret to enable pulling the example image from the registry.
    ```
-   kubectl create secret docker-registry bobs-books-repo-credentials \
+   $ kubectl create secret docker-registry bobs-books-repo-credentials \
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md:29]: Fixed.

File: examples/bobs-books/README.md [hunk 7]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
+
+   $ kubectl get gateway robert-ingress-rule-0-gw -n bobs-books -o jsonpath={.items[0].spec.servers[0].hosts[0]}
+   robert.bobs-books.11.22.33.44.xip.io
+
+   $ kubectl get gateway bobby-ingress-rule-0-gw -n bobs-books -o jsonpath={.items[0].spec.servers[0].hosts[0]}
+   bobby-front-end.bobs-books.11.22.33.44.xip.io
+   ```
+
 1. Access the applications. To access the applications in a browser, you will need to do one of the following:
-    * **Option 1:** Temporarily modify the `/etc/hosts` file (on Mac or Linux) or `c:\Windows\System32\Drivers\etc\hosts` file (on Windows 10), to add entries mapping the hosts used by the applications to the external IP address assigned to your gateway. For example:
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md:82]: Fixed.

File: examples/bobs-books/README.md [hunk 8]
```diff
@@ -62,38 +62,59 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
...
+    * **Option 1:** If you are using `xip.io` you can just access the applications using the generated host names, for example:
+
+      a. Robert's Books UI at `https://robert.bobs-books.11.22.33.44.xip.io/`.
+
+      b. Bobby's Books UI at `https://bobby-front-end.bobs-books.11.22.33.44.xip.io/bobbys-front-end`.
+
+      c. Bob's order manager  UI at `https://bobs-orders-wls.bobs-books.11.22.33.44.xip.io/bobs-bookstore-order-manager/orders`.
+
+    * **Option 2:** Temporarily modify the `/etc/hosts` file (on Mac or Linux) or `c:\Windows\System32\Drivers\etc\hosts` file (on Windows 10), to add entries mapping the hosts used by the applications to the external IP address assigned to your gateway. For example:
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md]: Fixed.

File: examples/bobs-books/README.md [hunk 9]
```diff
@@ -106,10 +127,10 @@ The Bob's Books example is a set of three applications based on WebLogic, Helido
     
 1. Verify that the application configuration, domains, coherence resources and ingress trait all exist.
```
[the remark this answers is not in the document]
Comment (fredtibbitts) [reply examples/bobs-books/README.md]: Fixed.

**Review by rosemarymarano — APPROVED** [review]

File: examples/sock-shop/README.md [hunk 10]
```diff
@@ -17,19 +17,19 @@ The Sock Shop application deployment artifacts are contained in the Verrazzano p
 
 1. Create a namespace for the Sock Shop application and add a label identifying the namespace as managed by Verrazzano.
    ```
-   kubectl create namespace sockshop
-   kubectl label namespace sockshop verrazzano-managed=true
+   $kubectl create namespace sockshop
```
Comment (kminder) [inline examples/sock-shop/README.md:20]: Space between $ and k?

File: examples/sock-shop/README.md [hunk 11]
```diff
@@ -53,41 +53,46 @@ ports, and such.
...
+   $ echo $HOST
+   sockshop-appconf.sockshop.11.22.33.44.xip.io
    ```
-   kubectl get service istio-ingressgateway -n istio-system
 
-   NAME                   TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)                      AGE
-   istio-ingressgateway   LoadBalancer   10.96.97.98   11.22.33.44   80:31380/TCP,443:31390/TCP   13d
+1. Get the `EXTERNAL_IP` address of the `istio-ingressgateway` service.
+   ```
+   $ ADDRESS=$(kubectl get service -n istio-system istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
```
Comment (kminder) [inline examples/sock-shop/README.md:65]: Would it make more sense to name this variable EXTERNAL_IP?

File: examples/springboot-app/README.md [hunk 12]
```diff
@@ -15,47 +15,55 @@ Install Verrazzano following the [installation instructions](https://verrazzano.
...
-1. By default, the application is deployed with a host value of `springboot.example.com`.
+1. Access the Springboot example application.
    There are several ways to access it:
 
    * **Using the command line**
      ```
-     curl -H "Host: springboot.example.com" http://11.22.33.44
-	 curl -H "Host: springboot.example.com" http://11.22.33.44/facts
+     $ curl -sk https://${HOST} --resolve ${HOST}:443:${ADDRESS}
+	  $ curl -sk https://${HOST}/facts --resolve ${HOST}:443:${ADDRESS}
```
Comment (kminder) [inline examples/springboot-app/README.md:55]: Remove extra spaces?

File: examples/todo-list/README.md [hunk 13]
```diff
@@ -144,27 +150,27 @@ The ToDo List application deployment artifacts are contained in the Verrazzano p
...
-   kubectl get pods -n todo-list
+   $ kubectl get pods -n todo-list
 
    NAME                     READY   STATUS    RESTARTS   AGE
    mysql-5c75c8b7f-vlhck    1/1     Running   0          19h
    tododomain-adminserver   2/2     Running   0          19h
    ```
 
-   Copyright (c) 2020, 2021, Oracle and/or its affiliates.
+Copyright (c) 2020, 2021, Oracle and/or its affiliates.
```
Comment (kminder) [inline examples/todo-list/README.md:176]: Is this supposed to be here?

Comment (rosemarymarano) [reply examples/todo-list/README.md:176]: @kminder I asked @markxnelson , "Quick question: should the copyright statement be in the example README?" His response, "doesn't hurt" My opinion is that whatever decisions we make, that we apply them consistently.

Comment (kminder) [reply examples/todo-list/README.md:176]: Also seemed strange that this is at the end of the file.

Comment (fredtibbitts) [reply examples/todo-list/README.md:176]: All but one of the READMEs already had these copyrights, I just added the one Rosemary noticed was missing and made them consistent.

File: examples/sock-shop/README.md (hunk 10 above)
Comment (fredtibbitts) [reply examples/sock-shop/README.md]: Fixed.

File: examples/springboot-app/README.md (hunk 12 above)
Comment (fredtibbitts) [reply examples/springboot-app/README.md]: Fixed.

**Review by kminder — APPROVED** [review]

**Pull request merged**